### ISSUE TITLE
Private interfaces are clickable but return a 404

### ISSUE THREAD

**Issue by nesk** [Kotlin/dokka#3299]
**Describe the bug**
I have a public class `Configuration` inheriting a private interface `ConfigurationInterface`: https://akkurate.dev/api/akkurate-core/dev.nesk.akkurate/-configuration/index.html

The interface name can be clicked, but it throws a 404 Not Found: https://akkurate.dev/api/akkurate-core/dev.nesk.akkurate/-configuration-interface/index.html

**Expected behaviour**
There are two options:
1. Hide the interface name, show `class Configuration` instead of `class Configuration : ConfigurationInterface`
2. Remove the link of the interface name

I think I would prefer the first option, there is no point to show the users that `Configuration` is inheriting `ConfigurationInterface`, **because it is private.**

**Screenshots**
| <img width="300" alt="image" src="https://github.com/Kotlin/dokka/assets/817508/ffbe787b-2b5d-4f8e-b567-6e0a11cca062">  | <img width="300" alt="image" src="https://github.com/Kotlin/dokka/assets/817508/79b2c690-783a-4d1d-bb14-5e8825a4a3c7"> |

**To Reproduce**
Basically:
```kotlin
interface ConfigurationInterface
class Configuration : ConfigurationInterface
```

You can find a production example here: https://github.com/nesk/akkurate/blob/1d74d2f4cdeef18eeddb424236a4f5aa824ce776/library/src/jvmMain/kotlin/dev/nesk/akkurate/Configuration.kt

**Dokka configuration**
_No specific configuration_

**Installation**
- Operating system: macOS
- Build tool: Gradle v8.1.1
- Dokka version: 1.9.0

**Additional context**
_None_

**Are you willing to provide a PR?**
Yes ✅

**Comment by nesk** [comment]
> @nesk, are you still willing to contribute?

Sorry for the late reply! Honestly, this isn't a top priority for me but, sure, I would love to if I find some time for this. Maybe the right is now, around christmas and new year, we'll see :)
